src/app/store/test-records/selectors/test-records.selectors.ts

```diff
@@ -44,6 +45,16 @@ export const selectedTestResultState = createSelector(
 
 export const testResultInEdit = createSelector(testResultsFeatureState, (state) => state.editingTestResult);
 
+export const cleanedTestResultInEdit = createSelector(testResultsFeatureState, (state) => {
```

Comment:
not sure this is the best way to approach this, as this is going to scatter logic which is removing data from the tech record before submission, think it makes more sense to follow the same path that we took for ADR stuff, which is to dispatch an action which calls a reducer which updates the tech record right before submission

Reply:
for reference, this action which is ADR specific, is dispatched on line 170 in the tech-record-summary-changes component

Reply:
I thought about this, but the only way to make this work is to refactor the test submission, and I was trying to keep the impact as minimal as possible. I could probably do it tbf but I'm not really sure how having the logic moved into a reducer is really any different?

Reply:
i was thinking purely from a consistency standpoint that its going to make it harder to debug if theres logic handling tech record additions/deletions before submissions which isn't centralised

what refactor would you need to do to test submission, would it not just be dispatching an action in the same block as the code i referenced above? or am i missing something obvious which makes this more complicated

Reply:
Thanks Phil, will update and reformat this and come back with another pass :)

Reply:
Thanks lads, updated!